examples/singleCell/LPdataAdaptor.cpp

```diff
@@ -20,310 +20,378 @@ using namespace std;
-    metadata->ArrayCentering = {vtkDataObject::CELL};
-    metadata->ArrayComponents = {1};
-    metadata->ArrayType = {VTK_FLOAT};
+    numMeshes = 2;
+    return 0;
+  }
+  //----------------------------------------------------------------------
+  int LPDataAdaptor::GetMeshMetadata(unsigned int id, sensei::MeshMetadataPtr &metadata) 
+  {
+    cout << "Calling GetMeshMetaData" << endl;
+    int rank, nRanks;
     
-    metadata->StaticMesh = 1;  
+    int nx = this->Internals->pb_nx;
+    int ny = this->Internals->pb_ny;
+    int nz = this->Internals->pb_nz; 	
+    MPI_Comm_rank(this->GetCommunicator(), &rank);
+    MPI_Comm_size(this->GetCommunicator(), &nRanks); 	
 
-    if (metadata->Flags.BlockDecompSet())
+    if (id == 0)
     {
-      metadata->BlockOwner.push_back(rank);
-      metadata->BlockIds.push_back(rank);
-    }
-  
-    metadata->BlockNumCells.push_back(this->Internals->nlocal/3);
-    metadata->BlockNumPoints.push_back(this->Internals->nlocal*3);
-    metadata->BlockCellArraySize.push_back(this->Internals->nlocal);
-  }
- 
-  else
-  {
-    metadata->MeshName = "fluid"; 
-	metadata->MeshType = VTK_IMAGE_DATA;
-	metadata->BlockType=VTK_IMAGE_DATA; 
-	metadata->CoordinateType = VTK_DOUBLE;
-	metadata->NumBlocks = nRanks;
-	metadata->NumBlocksLocal = {1}; 
-	metadata->NumArrays=1;
-	metadata->ArrayName = {"velocity"};
-	metadata->ArrayComponents = {3}; 
-	metadata->ArrayType = {VTK_DOUBLE};
-	metadata->ArrayCentering = {vtkDataObject::CELL};
-	metadata->StaticMesh = 1; 
+      //cout << "GetMeshMetaData Cells Test" << endl;
+      metadata->MeshName = "cells";
+      metadata->MeshType = VTK_MULTIBLOCK_DATA_SET; //VTK_POLY_DATA;
+      metadata->BlockType = VTK_POLY_DATA;
+      metadata->CoordinateType = VTK_DOUBLE;
+      metadata->NumBlocks = nRanks;
+      metadata->NumBlocksLocal = {1};
+      metadata->NumGhostCells = 0;
+      metadata->NumArrays = 0;
+      metadata->StaticMesh = 0;  
 
-    if (metadata->Flags.BlockDecompSet())
+      if (metadata->Flags.BlockExtentsSet())
+      {
+        //SENSEI_WARNING("lammps data adaptor. Flags.BlockExtentsSet()")
+        
+        // fixme
+        // There should be no extent for a PolyData, but ADIOS2 needs this
+        std::array<int,6> ext = { 0, 0, 0, 0, 0, 0};
+        metadata->Extent = std::move(ext);
+        metadata->BlockExtents.reserve(1);	// One block per rank
+        metadata->BlockExtents.emplace_back(std::move(ext));
+      }
+      
+      if (metadata->Flags.BlockDecompSet())
+      {
+        metadata->BlockOwner.push_back(rank);
+        metadata->BlockIds.push_back(rank);
+      }
+    
+      //We use nanglelist for BlockNumCells because it give the number of triangles on a given processor
+      metadata->BlockNumCells.push_back(this->Internals->nanglelist);
+      metadata->BlockNumPoints.push_back(this->Internals->nlocal + this->Internals->nghost );
+      metadata->BlockCellArraySize.push_back(0);
+    }
+    else

```

Comment:
I would replace this else with a specific comparison id == 1 to make this code future proof in case someone else adds more meshes

Reply:
Also add an error printout if id > 1

Reply:
I added an else statement. Will that work for if id > 1 or id is somehow null?